src/Microsoft.TestPlatform.Utilities/ClientUtilities.cs

```diff
@@ -54,6 +54,8 @@ public static void FixRelativePathsInRunSettings(XmlDocument xmlDocument, string
 
     private static void AddRunSettingsDirectoryNode(XmlDocument doc, string path)
     {
+        ValidateArg.NotNull(doc.DocumentElement, nameof(doc.DocumentElement));
```

Comment:
I'm not a big fan of this. This is going to throw `ArgumentNullException` while it's not the arg that is null but a field of it. IMO we should throw either `ArgumentException` or `InvalidOperationException` (I would go for the last one as I mentally associate it to "invalid state" but the first one is fine too).